internal/smtpsrv/conn.go

```diff
@@ -1047,6 +1048,39 @@ func (c *Conn) AUTH(params string) (code int, msg string) {
 	response := ""
 	if len(sp) == 2 {
```

Comment:
I think this condition needs to be adjusted cause current code would enter it with `LOGIN something`.

But I'd say it's probably a better idea to add a `Conn.LOGIN` method instead to make this kind of bug impossible. Common code could be factored out.

Reply:
Thanks for the quick response, yes. I'd be happy to refactor this into another method if the maintainer is OK with it.

Reply:
Thanks to both.

I agree it would be better to refactor this.

I would keep the AUTH function and then split lines 1046-1070 in a separate function for PLAIN, and call either that or a new LOGIN function based on the method the client is using.

Does that seem reasonable to you?